### ISSUE TITLE
Run bwc tests for the distribution (OpenSearch + plugins).

### ISSUE THREAD

**Issue by saratvemulapalli** [opensearch-project/opensearch-build#121]
Currently, each plugin has their own backward compatibility tests running in their GitHub actions. It sets up their own test cluster and test each individual subsequent tasks: mixed cluster test; rolling upgrade test; full cluster restart test... 
In plugin level, the current bwc tests tasks are running against the [test cluster framework](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/main/java/org/opensearch/gradle/testclusters) from the core team, which set up the test cluster with designated opensearch min version. Within the test cluster on plugin teams end, it only installs their own plugin and other required dependencies plugins.

We like to run backward compatibility tests once the distribution is ready. In another word, running the plugin BWC against the test clusters with all plugins present along with the OpenSearch min once we generate a new bundle distribution.

**Comment by peterzhuamazon** [comment]
This will be put on priority after the performance test.
@VachaShah and @gaiksaya please sync up and check what is still pending for the task.
Thanks.

**Comment by bbarani** [comment]
[Triage] We will revisit the design when we start working on BWC testing implementation. Closing this issue for now. We will re-open this again later if required.

**Comment by dblock** [comment]
We do need 1 issue that says "run bwc tests for every build of OpenSearch", which one do you want to leave open @bbarani?

**Comment by zelinh** [comment]
Closing in favor of https://github.com/opensearch-project/opensearch-build/issues/2870. Will update in that PR for progress.
